src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.openapi;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.swing.JFileChooser;
+import javax.swing.SwingUtilities;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpSender;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.openapi.converter.Converter;
+import org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter;
+import org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerException;
+import org.zaproxy.zap.extension.openapi.network.Requestor;
+import org.zaproxy.zap.extension.spider.ExtensionSpider;
+import org.zaproxy.zap.spider.parser.SpiderParser;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionOpenApi extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionOpenApi";
+
+    private static final String THREAD_PREFIX = "ZAP-Import-OpenAPI-";
+
+    private ZapMenuItem menuImportLocalOpenApi = null;
+    private ZapMenuItem menuImportUrlOpenApi = null;
+    private int threadId = 1;
+
+    private static final Logger LOG = Logger.getLogger(ExtensionOpenApi.class);
+
+    public ExtensionOpenApi() {
+        super(NAME);
+        this.setOrder(158);
```

Comment:
It's using the same order as an existing extension:
`Duplicate order 158 ExtensionImportWSDL/org.zaproxy.zap.extension.soap.ExtensionImportWSDL already registered, ExtensionOpenApi/org.zaproxy.zap.extension.openapi.ExtensionOpenApi will be added as an unordered extension`